src/client/java/teammates/client/scripts/scalabilitytests/InstructorCourseEnrollPageScaleTest.java

```diff
@@ -0,0 +1,93 @@
+import java.nio.file.Paths;
+
+/**
+ * Covers 'enroll' view for instructors.
+ * SUT: {@link InstructorCourseEnrollPage}.
+ */
+public class InstructorCourseEnrollPageScaleTest extends BaseUiTestCase {
+
+    private static final String DATA_FOLDER_PATH = "src/client/java/teammates/client/scripts/scalabilitytests/data/";
+    private InstructorCourseEnrollPage enrollPage;
+    private Logger logger = Logger.getLogger();
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/InstructorCourseEnrollPageUiTest.json");
+        removeAndRestoreDataBundle(testData);
+    }
+
+    @Test
+    public void testInstructorCourseEnrollPage() throws Exception {
+        testContent();
+        testSampleLink();
+
+        int[] loads = {10, 20, 50, 75, 100, 150};
+        testEnrollActionWithIncreasingLoad(loads);
+    }
+
+    private void testContent() throws Exception {
+
+        ______TS("typical enroll page");
+
+        AppUrl enrollUrl = createUrl(Const.ActionURIs.INSTRUCTOR_COURSE_ENROLL_PAGE)
+                .withUserId(testData.instructors.get("CCEnrollUiT.teammates.test").googleId)
+                .withCourseId(testData.courses.get("CCEnrollUiT.CS2104").getId());
+
+        enrollPage = loginAdminToPage(enrollUrl, InstructorCourseEnrollPage.class);
+
+        // This is the full HTML verification for Instructor Course Enroll Page, the rest can all be verifyMainHtml
+        enrollPage.verifyHtml("/instructorCourseEnrollPage.html");
+    }
+
+    private void testSampleLink() {
+
+        ______TS("link for the sample spreadsheet");
+        enrollPage.clickSpreadsheetLink();
+        String expectedTitle = "Course Enroll Sample Spreadsheet - Google Sheets";
+        enrollPage.verifyTitle(expectedTitle);
+    }
+
+    private void testEnrollActionWithIncreasingLoad(int[] loads) throws Exception {
+        for (int load : loads) {
+            ______TS("enroll action: " + load + " students");
+            testEnrollAction(load);
+            removeAndRestoreDataBundle(testData);
+        }
+    }
+
+    private void testEnrollAction(int numStudents) throws IOException {
+        AppUrl enrollUrl =
+                createUrl(Const.ActionURIs.INSTRUCTOR_COURSE_ENROLL_PAGE)
+                        .withUserId(testData.instructors.get("CCEnrollUiT.teammates.test").googleId)
+                        .withCourseId(testData.courses.get("CCEnrollUiT.CS2104").getId());
+
+        enrollPage = loginAdminToPage(enrollUrl, InstructorCourseEnrollPage.class);
+
+        String enrollString =
+                readData("InstructorCourseEnrollPageScaleTestData" + numStudents, Charset.defaultCharset());
+
+        logger.info("Testing with " + numStudents + " students...");
+        Stopwatch stopwatch = new Stopwatch();
+        stopwatch.start();
+        enrollPage.enroll(enrollString);
+        stopwatch.logTimeElapsedInSeconds(logger);
+    }
+
+    String readData(String filename, Charset encoding) throws IOException {
```

Comment:
The two methods are unique to their respective tests though (and rather different); it might be a better idea to wait until we have more scalability / performance tests to abstract the similar components.